ui/tools/checkversion.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to Apache Software Foundation (ASF) under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Apache Software Foundation (ASF) licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const fs = require("fs");
+
+const nvmVersion = fs.readFileSync(".node-version").toString().trim();
```

Comment:
I suppose we can impose version check by setting the `engine` field in `package.json`

https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines

Reply:
`engine` is better to place the version. I updated the script